runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1531,7 +1534,46 @@ NNFW_STATUS nnfw_session::train_export_circle(const char *path)
   if (!mmapfile.ensure_mmap())
     return NNFW_STATUS_ERROR;
 
-  // TODO: Update weights in mmapfile
+  try
+  {
+    _execution->iterateTrainableTensors([&](const onert::ir::OperandIndex &idx,
+                                            const onert::backend::train::ITrainableTensor *tensor) {
+      auto model = ::circle::GetModel(mmapfile.buf());
+      auto subgs = model->subgraphs();
+      if (!model || !subgs || subgs->size() != 1)
+        throw std::runtime_error("Circle does not has valid subgraph or has multiple subgraphs");
+
+      auto subg = subgs->Get(0); // Get 1st subgraph
+      if (!idx.valid() || idx.value() >= subg->tensors()->size())
+        throw std::runtime_error("Trainable tensor index is out of range");
+
+      auto buf_idx = subg->tensors()->Get(idx.value())->buffer();
+      const ::circle::Buffer *buffer = (*model->buffers())[buf_idx];
+      if (!buffer || !buffer->data())
+        throw std::runtime_error("Buffer for trainable tensors is invalid");
+
+      const flatbuffers::Vector<uint8_t> *array = buffer->data();
+      if (!array)
+        throw std::runtime_error("Data for trainable tensor's buffer is invalid");
+
+      if (!array->Data())
+        throw std::runtime_error("Data for trainable tensor's buffer is invalid");
+
+      auto org_buf_sz = array->size();
+      if (org_buf_sz != tensor->total_size())
+        throw std::runtime_error("Trained tensor buffer size does not match original tensor's one");
+
+      memcpy(const_cast<uint8_t *>(array->Data()), tensor->buffer(), org_buf_sz);
```

Comment:
```suggestion
      memcpy(const_cast<uint8_t *>(org_buf), tensor->buffer(), org_buf_sz);
```

Or you may want this.

Reply:
Yes, right. I think this change is better than before. :)